zarr/util.py

```diff
@@ -286,9 +287,9 @@ def normalize_order(order: str) -> str:
     return order
 
 
-def normalize_dimension_separator(sep: Optional[str]) -> Optional[str]:
+def normalize_dimension_separator(sep: Optional[str]) -> Optional[DIMENSION_SEPARATOR]:
     if sep in (".", "/", None):
-        return sep
+        return cast(DIMENSION_SEPARATOR, sep)
```

Comment:
If `sep` is `None`, then I think this cast is incorrect?

Reply:
👍 I fixed the casting